lib/Modules/ADHooks/ADHooks.psm1

```diff
@@ -1458,3 +1550,52 @@ function Invoke-ADSPNRelationChangedHook {
         }
     }
 }
+
+function Invoke-UpdateStatusHook {
+    if(!(Confirm-Leader)){
```

Comment:
avoid nesting if possible. You can drop the else, considering that you exit if the if is satisfied.